.bazelrc

```diff
@@ -1,16 +1,39 @@
+# share bazel cache between checkouts of the same project
+# and keep old build caches around for longer
+build --disk_cache=~/.cache/shared_bazel_action_cache
+build --repository_cache=~/.cache/shared_bazel_repository_cache
+
+# better caching / reproducibility
+build --incompatible_strict_action_env=true
+build --experimental_output_directory_naming_scheme=diff_against_baseline
+
+# disable automatic toolchain detection for C/C++
 build --incompatible_enable_cc_toolchain_resolution
 build --action_env BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1
-build --platforms @zig_sdk//libc_aware/platform:linux_amd64_gnu.2.34
+
+# disable automatic toolchain detection for Java
 build --incompatible_use_toolchain_resolution_for_java_rules
 build --java_runtime_version=pinned_remotejdk_11
 build --tool_java_runtime_version=pinned_remotejdk_11
-build --experimental_output_directory_naming_scheme=diff_against_baseline
-test --java_runtime_version=pinned_remotejdk_11
-test --tool_java_runtime_version=pinned_remotejdk_11
-test --incompatible_use_toolchain_resolution_for_java_rules
-test --experimental_output_directory_naming_scheme=diff_against_baseline
 
+# inject version information into binaries
+build --stamp --workspace_status_command=tools/workspace_status
+
+# compile for linux_amd64 by default (this is the target for any binaries that go into the cluster)
+build --platforms @zig_sdk//libc_aware/platform:linux_amd64_gnu.2.34
+
+# enable tpm simulator for tests
+test --//bazel/settings:tpm_simulator
+
+# disable test caching (rerun all test cases even if they passed before)
+test --cache_test_results=no
```

Comment:
Why disable caching for tests globally?

Reply:
This is the current behaviour for our go unit tests and it only reruns the tests (it doesn't force recompilation, only re-execution). Without this flag, running `bazel test ...` twice will only run the tests the first time. Not sure what is better developer UX in this case.

Reply:
Mh, actually it seems to cache the build steps anyway (cannot test fully since tests seem to be broken on macOS). If it does and doesn't build everything from scratch then I guess that's fine.